### ISSUE TITLE
An error when using DCNv2

### ISSUE THREAD

**Issue by WangLibo1995** [open-mmlab/mmdetection#7389]
I test the DCNv2, a error raised:
![image](https://user-images.githubusercontent.com/38718224/158126984-cdc44c5d-b165-4fa3-836c-6993e155c8a6.png)
Any error in my code? How to correct it?

**Comment by WangLibo1995** [comment]
I check the mmcv, it seems not support dcnv2.

from torch import nn

from .registry import CONV_LAYERS

**CONV_LAYERS.register_module('Conv1d', module=nn.Conv1d)
CONV_LAYERS.register_module('Conv2d', module=nn.Conv2d)
CONV_LAYERS.register_module('Conv3d', module=nn.Conv3d)
CONV_LAYERS.register_module('Conv', module=nn.Conv2d)**


def build_conv_layer(cfg, *args, **kwargs):
    """Build convolution layer.
    Args:
        cfg (None or dict): The conv layer config, which should contain:
            - type (str): Layer type.
            - layer args: Args needed to instantiate an conv layer.
        args (argument list): Arguments passed to the `__init__`
            method of the corresponding conv layer.
        kwargs (keyword arguments): Keyword arguments passed to the `__init__`
            method of the corresponding conv layer.
    Returns:
        nn.Module: Created conv layer.
    """
    if cfg is None:
        cfg_ = dict(type='Conv2d')
    else:
        if not isinstance(cfg, dict):
            raise TypeError('cfg must be a dict')
        if 'type' not in cfg:
            raise KeyError('the cfg dict must contain the key "type"')
        cfg_ = cfg.copy()

    layer_type = cfg_.pop('type')
    **if layer_type not in CONV_LAYERS:
        raise KeyError(f'Unrecognized layer type {layer_type}')**
    else:
        conv_layer = CONV_LAYERS.get(layer_type)

    layer = conv_layer(*args, **kwargs, **cfg_)

    return layer

**Comment by Czm369** [comment]
Maybe you can update mmcv==1.4.6

**Comment by WangLibo1995** [comment]
> Maybe you can update mmcv==1.4.6

![image](https://user-images.githubusercontent.com/38718224/158132561-8a5fd7c5-3b98-4bac-a636-d85f45c882e3.png)

**Comment by WangLibo1995** [comment]
> Maybe you can update mmcv==1.4.6

The error is still there.

**Comment by Czm369** [comment]
DCNv2 is not in the CONV_LAYERS

**Comment by WangLibo1995** [comment]
> DCNv2 is not in the CONV_LAYERS



> Maybe you can try from mmcv.ops import ModulatedDeformConv2dPack dcn = ModulatedDeformConv2dPack( ... 1, ... 1, ... kernel_size=(2, 2), ... stride=1, ... padding=1, ... deform_groups=1, ... bias=False)

Many thanks for your help! It's ok now.
